packages/fxa-settings/src/models/Account.ts

```diff
@@ -530,6 +530,61 @@ export class Account implements AccountData {
     }
   }
 
+  async verifyPasswordForgotToken(token: string, code: string) {
+    try {
+      const verifyCodeResult = await this.apolloClient.mutate({
+        mutation: gql`
+          mutation passwordForgotVerifyCode(
+            $input: PasswordForgotVerifyCodeInput!
+          ) {
+            passwordForgotVerifyCode(input: $input) {
+              accountResetToken
+            }
+          }
+        `,
+        variables: { input: { token, code } },
+      });
+      return verifyCodeResult.data.passwordForgotVerifyCode;
+    } catch (err) {
+      const errno = (err as ApolloError).graphQLErrors[0].extensions?.errno;
+      if (errno && AuthUiErrorNos[errno]) {
+        throw AuthUiErrorNos[errno];
+      }
+      throw AuthUiErrors.UNEXPECTED_ERROR;
+    }
```

Comment:
Doesn't have to be done here but almost seems like we could use some generic error handler here since we have this code a few times in this file 🤔